resources/tools.rb

```diff
@@ -19,9 +19,35 @@
 # limitations under the License.
 #
 
-actions :install
+resource_name 'octopus_deploy_tools'
+
+property :path, kind_of: String, name_attribute: true
+property :source, kind_of: String, required: true
+property :checksum, kind_of: String
+
 default_action :install
 
-attribute :path, kind_of: String, default: 'C:\\octopus', name_attribute: true
-attribute :source, kind_of: String, required: true
-attribute :checksum, kind_of: String
+action :install do
+  path = new_resource.path
```

Comment:
I think it would be better not to define these. You can refer to `path` without having to use `new_resource.path` anyway, the only instances where you have to use the `new_resource.x` construct are where you are trying to use it in a resource that has a property `x`.

This is the situation below where you have
```
checksum checksum if checksum
```

so you will probably run into problems there